UnityProject/Assets/Scripts/Systems/Fluids/MixAndVolume.cs

```diff
@@ -21,9 +21,18 @@ public float InternalEnergy
 			get => mix.InternalEnergy + gasMix.InternalEnergy;
 			set
 			{
+
+				if (float.IsNormal(value) == false && value != 0)
+				{
+					Logger.LogError($"AAAAAAAAAAAAA REEEEEEEEE MixAndVolume InternalEnergy Invalid number!!!! {value}");
```

Comment:
I know you find it funny, but please make a descriptive error so we, the other developers, understand what is happening when we see this stuff in the log.

Reply:
MixAndVolume InternalEnergy Invalid number!!!! {value} 
is a pretty descriptive log